includes/functions/functions_search.php

```diff
@@ -98,39 +98,46 @@ function zen_parse_search_string($search_str = '', &$objects = array()) {
 // Move on to the next word
                     $k++;
                     continue;
-                } else {
-                    /* If the $piece ends in double quotes, strip the double quotes, tack the
-                       $piece onto the tail of the string, push the $tmpstring onto the $haves,
-                       kill the $tmpstring, turn the $flag "off", and return.
-                    */
-                    $tmpstring .= ' ' . trim(preg_replace('/"/', ' ', $pieces[$k]));
-
-// Push the $tmpstring onto the array of stuff to search for
-                    $objects[] = trim($tmpstring);
-
-                    for ($j=0, $n=count($post_objects); $j<$n; $j++) {
-                        $objects[] = $post_objects[$j];
-                    }
+                }
+                /* If the $piece ends in double quotes, strip the double quotes, tack the
+                   $piece onto the tail of the string, push the $tmpstring onto the $haves,
+                   kill the $tmpstring, turn the $flag "off", and return.
+                */
+                $tmpstring .= ' ' . trim(preg_replace('/"/', ' ', $pieces[$k]));
 
-                    unset($tmpstring);
+// Push the $tmpstring onto the array of stuff to search for, trimming again, in case the above
+// statement added a leading space.
+                $objects[] = trim($tmpstring);
 
-// Turn off the flag to exit the loop
-                    $flag = 'off';
+                for ($j = 0, $n = count($post_objects); $j < $n; $j++) {
+                    $objects[] = $post_objects[$j];
                 }
+
+                unset($tmpstring);
+
+// Turn off the flag to exit the loop
+                break;
             }
         }
     }
 
 // add default logical operators if needed
-    $temp = array();
```

Comment:
Something in this file's refactoring is breaking the search functionality.

Example: searching for `graphics card` used to work before this refactoring; now it requires `graphics and card` to bring results.

Reply:
Revised line 149 was the issue:
````diff
-    for ($i = 0; $i < $j + 1; $i++) {
+    for ($i = 0; $i < $j + 2; $i++) {
````
But again, whatever.